presentation/common/src/presentation-common/hierarchy/Node.ts

```diff
@@ -65,17 +67,39 @@ export interface NodeJSON {
   isCheckboxVisible?: boolean;
   isChecked?: boolean;
   isCheckboxEnabled?: boolean;
-  extendedData?: { [key: string]: any };
+  extendedData?: {
+    [key: string]: any;
+  };
 }
 
+/** @alpha */
+export type PartialNode = AllOrNone<Partial<Node>, "key" | "label">;
+/** @alpha */
+export type PartialNodeJson = AllOrNone<Partial<NodeJSON>, "key" | "labelDefinition">;
+type AllOrNone<T, P extends keyof T> = Omit<T, P> & ({ [K in P]?: never } | Required<Pick<T, P>>);

```

Comment:
`AllOrNone` is intentionally not exported.

Reply:
but you're using it in exported types... afaik that's invalid

Reply:
The code and documentation compile as expected, `AllOrNone` is redeclared in module output and `PartialNode` type works as expected when imported by a consuming library outside this monorepo. I don't understand what's invalid about it.

Reply:
I'm surprised that documentation compiles - maybe it's because the types are `@alpha` and don't get included in the docs?.. Otherwise what would be the definition of `PartialNode` in there, if the docs know nothing about `AllOrNone`? That's exactly what can be seen in `presentation-common.api.md` - `PartialNode` and `PartialNodeJson` are defined using `AllOrNone`, but the latter is not included in the markdown.

Reply:
Maybe I'm wrong though... @calebmshafer , @williamkbentley any ideas?

Reply:
You're right @grigasp, from the docs perspective `PartialNode` is `@alpha` so it is totally excluded.

Reply:
Yes, `@alpha` items are excluded from the docs -- that's why I made them `@public` and added dummy description as a test.

Reply:
@calebmshafer, do you have an opinion whether it's correct to **not** export a `type` when it's used by exported types?